crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -1182,6 +1182,103 @@ impl BeaconState {
 
         Ok(())
     }
+
+    pub fn process_justification_and_finalization(&mut self) -> anyhow::Result<()> {
+        // Initial FFG checkpoint values have a `0x00` stub for `root`.
+        // Skip FFG updates in the first two epochs to avoid corner cases that might result in
+        // modifying this stub.
+        if self.get_current_epoch() <= GENESIS_EPOCH + 1 {
+            return Ok(());
+        }
+        let previous_indices = self.get_unslashed_participating_indices(
+            TIMELY_TARGET_FLAG_INDEX,
+            self.get_previous_epoch(),
+        )?;
+        let current_indices = self.get_unslashed_participating_indices(
+            TIMELY_TARGET_FLAG_INDEX,
+            self.get_current_epoch(),
+        )?;
+        let total_active_balance = self.get_total_active_balance();
+        let previous_target_balance = self.get_total_balance(previous_indices);
+        let current_target_balance = self.get_total_balance(current_indices);
+        self.weigh_justification_and_finalization(
+            total_active_balance,
+            previous_target_balance,
+            current_target_balance,
+        )?;
+        Ok(())
+    }
+
+    pub fn weigh_justification_and_finalization(
+        &mut self,
+        total_active_balance: u64,
+        previous_epoch_target_balance: u64,
+        current_epoch_target_balance: u64,
+    ) -> anyhow::Result<(), anyhow::Error> {
+        let previous_epoch = self.get_previous_epoch();
+        let current_epoch = self.get_current_epoch();
+        let old_previous_justified_checkpoint = self.previous_justified_checkpoint;
+        let old_current_justified_checkpoint = self.current_justified_checkpoint;
+
+        // Process justifications
+        self.previous_justified_checkpoint = self.current_justified_checkpoint;
+        for i in (1..JUSTIFICATION_BITS_LENGTH as usize).rev() {
```

Comment:
```suggestion
        for i in (1..JUSTIFICATION_BITS_LENGTH as usize) {
```

isn't reverse not needed?